File: 0145_Binary Tree Postorder Traversal/BinaryTreePostorderTraversal.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public IList<int> PostorderTraversal(TreeNode root) {
        var list = new List<int>();
        if(root == null) return list;
        var stack = new Stack<TreeNode>();
        stack.Push(root);
        while(stack.Count > 0)
        {
            var node = stack.Pop();
            list.Add(node.val);
            if(node.left!=null) stack.Push(node.left);
            if(node.right!=null) stack.Push(node.right);
        }
        
        list.Reverse();
        return list;
    }
}

File: 0145_Binary Tree Postorder Traversal/BinaryTreePostorderTraversal.java
class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
        List<Integer> list = new LinkedList<Integer>();
        if(root == null) return list;
        Stack<TreeNode> left = new Stack<TreeNode>();
        Stack<TreeNode> right = new Stack<TreeNode>();

        left.add(root);

        while(left.size()>0){
            TreeNode node = left.pop();
            right.add(node);

            if(node.left!=null) left.add(node.left);
            if(node.right!=null) right.add(node.right);
        }
            while(right.size() > 0){
                list.add(right.pop().val);
            }
        }
    }
}